src/coreclr/src/gc/gc.cpp

```diff
@@ -4014,23 +4017,51 @@ BOOL gc_heap::reserve_initial_memory (size_t normal_size, size_t large_size, siz
         return FALSE;
     }
 
-    size_t requestedMemory = memory_details.block_count * (normal_size + large_size + pinned_size);
+    size_t temp_pinned_size = (separated_poh_p ? 0 : pinned_size);
+    size_t requestedMemory = memory_details.block_count * (normal_size + large_size + temp_pinned_size);
 
     uint8_t* allatonce_block = (uint8_t*)virtual_alloc (requestedMemory, use_large_pages_p);
+    uint8_t* separated_poh_block = nullptr;
+    if (allatonce_block && separated_poh_p)
+    {
+        separated_poh_block = (uint8_t*)virtual_alloc ((memory_details.block_count * pinned_size), false);
```

Comment:
>memory_details.block_count * pinned_size [](start = 56, length = 40)

nit - can we make `memory_details.block_count * pinned_size` a variable so you don't have to calculate it again a few lines down below?